libs/barista-components/quick-filter/src/quick-filter-group.ts

```diff
@@ -235,8 +235,12 @@ export class DtQuickFilterGroup<T = any> implements AfterViewInit {
       );
       if (!this.isDetail && items.length > this.maxGroupItems) {
         this._truncatedGroupItems = true;
-        this._showMoreCount = items.length - this.maxGroupItems;
-        return items.slice(0, this.maxGroupItems);
+        const filteredItems = items.filter(
+          (item, idx) => idx <= this.maxGroupItems || this._isActive(item),
```

Comment:
Please do not abbreviate variable names where it is not entirely clear on what the variable should be.
See [Coding standards](https://github.com/dynatrace-oss/barista/blob/master/CODING_STANDARDS.md#naming)

```suggestion
          (item, index) => index <= this.maxGroupItems || this._isActive(item),
```